great_expectations/experimental/datasources/data_asset/data_connector/data_connector.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import annotations
+
+import logging
+import pathlib
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+from great_expectations.core.id_dict import BatchSpec
+
+if TYPE_CHECKING:
+    from great_expectations.core.batch import BatchDefinition
+    from great_expectations.experimental.datasources.interfaces import BatchRequest
+
+
+logger = logging.getLogger(__name__)
+
+
+# noinspection SpellCheckingInspection
+class DataConnector:
```

Comment:
This class should be ABC.

Reply:
@donaldheppner I really like this suggestion -- I am going to try it and notify you once it has been committed.  Thanks.

Reply:
@donaldheppner This has been completed -- committed.  Thanks.